### PR TITLE
test(electron): validate packaged MCP sidecar

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
Adds Electron Windows/Linux package validation for the bundled OpenDucktor MCP sidecar and tightens the MCP host/tool contracts that the packaged sidecar exposes at runtime.

## Goal
The obsolete web-host artifact path is no longer the release path that needs validation. This PR makes the Electron package build fail fast when Windows or Linux artifacts do not contain the OpenDucktor MCP sidecar at the resources path expected by the TypeScript host, while keeping external tools such as `bd`, `dolt`, and `opencode` unbundled.

## Context
Packaged Electron startup sets the host environment from `process.resourcesPath`. The TypeScript host then resolves the bundled MCP sidecar through `OPENDUCKTOR_OPENDUCKTOR_MCP_PATH` and `OPENDUCKTOR_BUNDLED_BIN_DIR`. Without package-level validation, a release could build successfully while omitting the sidecar or shipping it without executable permissions on Unix.

During final PR verification, the root Bun test command exposed that the Builder session's `ODT_ALLOWED_TOOLS` environment intentionally hides `odt_get_workspaces`, which is invalid for the public MCP test suite. The PR therefore also makes the MCP listed-tool contract explicit for zero-input tools and structured output schemas, so client-visible metadata matches the tool result contract.

## Technical Choices
- Added an Electron package verification step after Electron Builder completes and before artifact publication.
- Validated `resources/bin/openducktor-mcp.exe` for Windows and `resources/bin/openducktor-mcp` for Linux, including non-empty payload checks and Unix executable-bit checks.
- Kept macOS and unrelated package targets out of this validation path.
- Preserved packaged Electron environment ownership in `electron-process-environment.ts`, with tests covering Windows and Linux resource paths.
- Rejected directory paths in host executable validation before checking platform-specific runnability.
- Registered MCP output schemas and explicit empty input properties for zero-argument tools so `structuredContent` remains a documented, client-visible contract.

## Verification
- `rtk bun run lint`
- `rtk bun run typecheck`
- `rtk env -u ODT_ALLOWED_TOOLS -u ODT_WORKSPACE_ID -u ODT_HOST_TOKEN -u ODT_HOST_URL -u ODT_FORBID_WORKSPACE_ID_INPUT bun run test`
- `rtk bun run build`
- `rtk cargo fmt --all --check` in `apps/desktop/src-tauri`
- `rtk cargo clippy --workspace --all-targets -- -D warnings` in `apps/desktop/src-tauri`
- `rtk cargo check --workspace --all-targets --locked` in `apps/desktop/src-tauri`
- `rtk cargo test --workspace --locked` in `apps/desktop/src-tauri` (`984 passed, 1 ignored`)
- `rtk cargo build --workspace --locked` in `apps/desktop/src-tauri`
- `rtk git rev-list --left-right --count origin/main...HEAD` returned `0 4`; no rebase was needed.

## Summary by CodeRabbit

* **New Features**
  * Added verification for bundled MCP sidecar packages in Electron releases
  * Added output schema metadata for MCP tool definitions

* **Bug Fixes**
  * Improved executable file validation to reject directories with execute permissions
  * Enhanced process cleanup handling with better error recovery

* **Tests**
  * Added comprehensive sidecar package verification test suite
  * Expanded MCP command validation tests with edge cases
  * Updated platform-specific test configurations

[Review Change Stack](https://app.coderabbit.ai/change-stack/Maxsky5/openducktor/pull/608?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: apps/electron/scripts/package-build.ts [hunk 1]
```diff
@@ -220,6 +221,11 @@ export const buildElectronPackage = async ({
     env: resolveElectronBuilderEnv(signed, process.env),
   });
 
+  const verifiedSidecarPath = await verifyPackagedMcpSidecar({ platform, releaseDirectory });
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/electron/scripts/package-build.ts]: Thanks. I am intentionally keeping this verifier target-platform based. It validates the packaged target artifact layout and sidecar filename, not the host-side preparation workspace. If a cross-platform package contains a host-format sidecar, that package is invalid and should fail validation instead of being accepted. Switching this check to host platform semantics would mask the broken release artifact.
